src/dex/sundaeswap-v3.ts

```diff
@@ -0,0 +1,238 @@
+import { LiquidityPool } from './models/liquidity-pool';
+import { BaseDataProvider } from '@providers/data/base-data-provider';
+import { Token } from './models/asset';
+import { BaseDex } from './base-dex';
+import {
+    AssetBalance,
+    DatumParameters,
+    DefinitionConstr,
+    DefinitionField,
+    PayToAddress,
+    RequestConfig, SpendUTxO,
+    SwapFee,
+    UTxO
+} from '@app/types';
+import { DefinitionBuilder } from '@app/definition-builder';
+import { correspondingReserves, tokensMatch } from '@app/utils';
+import { AddressType, DatumParameterKey } from '@app/constants';
+import pool from '@dex/definitions/sundaeswap-v1/pool';
+import order from '@dex/definitions/sundaeswap-v1/order';
+import { BaseApi } from '@dex/api/base-api';
+import { Script } from 'lucid-cardano';
+import { SundaeSwapV3Api } from '@dex/api/sundaeswap-v3-api';
+
+export class SundaeSwapV3 extends BaseDex {
+
+    public static readonly identifier: string = 'SundaeSwapV1';
+    public readonly api: BaseApi;
+
+    /**
+     * On-Chain constants.
+     */
+    // TODO - update for mainnet
+    public readonly orderAddress: string = 'addr_test1zpyyj6wexm6gf3zlzs7ez8upvdh7jfgy3cs9qj8wrljp925y2usavetr7qg4zg6cc8kqup4veczmfykgquw2kmmuhs5qkzsen3';
+    public readonly poolAddress: string = 'addr_test1wzq5p39cjs50cfjwjzcscuw98fxrl8899dnkhuwek50tnjs09luug';
+    public readonly lpTokenPolicyId: string = '4086577ed57c514f8e29b78f42ef4f379363355a3b65b9a032ee30c9';
+    public readonly validityTokenPolicyId: string = '8140c4b89428fc264e90b10c71c53a4c3f9ce52b676bf1d9b51eb9ca';
+    public readonly cancelDatum: string = 'd87a80';
+    public readonly orderScript: Script = {
+        type: 'PlutusV1',
```

Comment:
Should be `PlutusV2`, I think. cc @Quantumplation

Reply:
Changed to PlutusV2 in 9a31ee95cb7024e03f3da20e4e91deea8cb301bd